### ISSUE TITLE
Add `AsCollection` option to the `-Split` Operator

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#18926]
### Summary of the new feature / enhancement

Currently the `-Split` operator returns a string array (`String[]`) which immediately unrolls which is not always useful in case you want change the returned strings in a larger stream:

Consider the following list:
```PowerShell
$List = 'a,b', 'c,d', 'e,f'
```
I would like split each item on the comma and add the numbers `1` and `2` in the front of the resulted characters:
```Console
1a
2b
1c
2d
1e
2f
```

Probably the easiest way to do this is reassigning the split item:
* create a loop
* split each item and **reassign the results**
* make changes to each subitem in the assignment

```PowerShell
$List | Foreach-Object {
    $Split = $_ -Split ','
    '1' + $Split[0]
    '2' + $Split[1]
}
```
...or do things along with the format string operator:
```PowerShell
$List | Foreach-Object {
    $Split = $_ -Split ','
    '1{0} 2{1}' -f $Split
}
```

### Proposed technical implementation details (optional)

If `-Split` would optionally return a `[Collections.ObjectModel.Collection[String]]`, I could have the following syntactic sugar (wishful thinking):
```PowerShell
$List -Split ',',0,'AsCollection' | Foreach-Object {
    '1' + $_[0]
    '2' + $_[1]
}
```
In other words:
```PowerShell
$SplitAsCollection = # $List -Split ',',0,'AsCollection'
[Collections.ObjectModel.Collection[String]]('a', 'b'),
[Collections.ObjectModel.Collection[String]]('c', 'd'),
[Collections.ObjectModel.Collection[String]]('e', 'f')
```

```PowerShell
$SplitAsCollection | Foreach-Object { # $List -Split ',',0,'AsCollection' | Foreach-Object { 
    '1' + $_[0]
    '2' + $_[1]
}
```
...or do things along with the format string operator:
```PowerShell
# $List -Split ',',2,'AsCollection' | Foreach-Object { '1{0} 2{1}' -f @($_) }
$SplitAsCollection | Foreach-Object { '1{0} 2{1}' -f @($_) }
```
(where the `-f` operator should actually recognize a [psobject] collection: `'1{0} 2{1}' -f $_`, see: #14355)

**Comment by 237dmitry** [comment]
I would be careful with such requests. A million scripts have been written where the `-split` operator returns an array of strings. What could be the consequences can not be foreseen.

> Besides, the current syntax (as in the example in the summary) is still supposed to work on iterated scalars

This is the solution for your specific case.

**Comment by iRon7** [comment]
@237dmitry,

You're probably right, I just want to express the thought but it will probable break to many things.
I have changed the angle of the request to **Add AsCollection option to the -Split Operator**.

**Comment by 237dmitry** [comment]
> Add AsCollection option to the -Split Operator.

This is a completely different matter. I fully support you.

**Comment by jhoneill** [comment]
It might be better to have a new operator, currently 
`$list -split ",","1"`  is valid, and the either you'd to write need `$list -split ",",0,"AsCollection" ` or the parser would need to figure out that `$list -split ",","AsCollection"` ,meant that. 

**This is a useful thing to have**. But I don't like  `-operator  operand , "magicword"`  as syntax. If we were calling it as function it would be  `split  $list -Ascollection` and the behavioural switch is parameter not a special value for an operand.  

Just as we have `-cswitch` and `-iswitch`  and not `$list -split ",","CaseSensitive"` , this should be `-collectionSplit` or something similar but more elegant.

**Comment by SeeminglyScience** [comment]
It would be helpful to show some real world examples where this would be useful. Especially some that would be common enough to warrant adding a new syntax vs just piping to `ForEach-Object` as shown in the workaround. Bonus points for existing modules that would benefit.

---

I do also worry about the readability. It's a little hard to tell what the proposal does, but the workaround of piping to `ForEach-Object` is very clear.

**Comment by iRon7** [comment]
@SeeminglyScience ,

> show some real world examples

The request/thought started from this StackOverflow answer: https://stackoverflow.com/a/75079811/1701026

But in general, I would like to state: anywhere where there is a collection on the LHS, it is probably more convenient to have a stream of collections returned than a list of single items:

Anywhere where you would use [ConvertFrom-StringData](https://learn.microsoft.com/powershell/module/microsoft.powershell.utility/convertfrom-stringdata), but more flexibel on the delimiter:
```PowerShell
$ht = @{}
@(
    'a := 1'
    'b := 2'
    'c := 3'
) -Split '\s*\:\=\s*',2,'AsCollection' | Foreach-Object { $ht[$_[0]] = $ht[$_[1]] }
```

Rather than using `Import-Csv` for something like a header-less `|` delimited file:
```PowerShell
Import-Csv .\Test.txt -Header @(0..?) -Delimiter '|' |ForEach-Object { Write-Host $_.0 ...
```
Do:
```PowerShell
Get-Content .\Test.txt -Split '|',0,'AsCollection' | ForEach-Object { Write-Host $_[0] ...
```

(Note that the original title of this issue was "**`-Split` should return a `collection`**" where I would actually expect this to be the default behavior but changing this now will obviously break too much)

Anyways, I think that the "*some real world example*" are indeed quiet limited because there are already many ways to workaround this proposal.
